python/rmm/_version.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib.resources
+
+__version__ = (
+    importlib.resources.files("rmm").joinpath("VERSION").read_text().strip()
```

Comment:
I don't really like that importing the package now requires reading another file, just to populate a version string. Is "single-source" the only benefit of the VERSION approach?

Reply:
Assuming that you include the dynamic version metadata support in pyproject.toml as part of single-sourcing (and not just sharing across multiple packages), yes that is the primary benefit. It is the only approach to single-sourcing that is compatible with the range of different build processes that we're currently supporting. I agree that it's not perfect, but given the number of iterations that we've gone through with these PRs across all of RAPIDS I would prefer to leave this as is for now. Every alternative has tradeoffs, so there isn't a clear winner, and at least in the specific case of how we minimize duplication of the version we're going to have to revisit this again in the near future once we switch over to scikit-build-core.